### PR TITLE
feat(API): retrieve transactionAnchor info in health report

### LINKED ISSUE #832: Add a Health Check for Anchor Retries
## Goal

We need a way to track how often we are resubmitting batches before they become part of the blockchain.

## Reference

RDD: [Blockchain Confirmations](https://github.com/poetapp/documentation/blob/master/rdd/blockchain-confirmations.md#blockchain-confirmations)

Parent Issue: #707 

## Implementation Details
* [x] Have the health module listen to `ipfsHashTxId` event
* [x] Upsert the ipfsHash and the current txn id in a separate collection internal to the Health module. If the action results in an insert, set attempts = 1. If the txn id has changed increment attempts by 1. Record timestamp of upsert
* [x] on getHealth, report the number of batches with attempts > 1, grouped by attempts. Store result in health collection.
* [x] listen for blockDownloaded, purge db entries with the txIds.

## Out of Scope

* Managing a potential race condition between blockchainWriter and blockchainReader and getHealthCheck.

### PR DESCRIPTION
**Pull request by krobi64**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

Resolves #832, #707 

1. include the transactionAnchorRetryInfo from the health collection in the `GET /health` API call
2. verify the health report in the functional test

### CODE REVIEW

File: src/API/HealthController.ts [hunk 1]
```diff
@@ -1,36 +1,48 @@
...
 export interface Dependencies {
   readonly db: Db
+  readonly logger: Pino.Logger
 }
 
 export interface Arguments {
   readonly dependencies: Dependencies
 }
 
+const emptyTransactionAnchorRetryInfo: { transactionAnchorRetryInfo: TransactionAnchorRetryInfo } = {
```
Comment (kennylavender) [inline src/API/HealthController.ts]: formatting this line to be a bit shorter would be nice.

Comment (krobi64) [reply src/API/HealthController.ts]: should we modify our ts-lint rules to enforce shorter lines?

Comment (krobi64) [reply src/API/HealthController.ts]: I created an interface to shorten the line

Comment (kennylavender) [reply src/API/HealthController.ts]: lol :)

I think prettier originally kept our lines short, looks like it when it was removed we increased it to 120 https://github.com/poetapp/tslint-rules/blame/b0b654f6419a5a9c75f8a0991ecabe42186c71ee/poet-tslint.json#L33

> I created an interface to shorten the line

Thanks :D

Comment (krobi64) [reply src/API/HealthController.ts]: We should reduce it if prettier was keeping it shorter

File: src/API/HealthController.ts [hunk 2]
```diff
@@ -100,6 +116,30 @@ export class HealthController {
     }
   }
 
+  private async getTransactionRetryInfo(): Promise<TransactionAnchorRetryInfo> {
+    this.logger.child({ message: 'getTransactionRetryInfo' })
```
[the remark this answers is not in the document]
Comment (kennylavender) [reply src/API/HealthController.ts]: Oops, didn't even notice this changes, so used to see these logger everywhere I must have glanced over them

**Pull request merged**